### PR TITLE
Add support for equi-rectangular skymaps (2 by 1)

### PR DESCRIPTION
**Pull request by samcake**
By popular demand from myself, we wanted to support the 2 x 1 aspect ratio skymap which is an equi-rectangular projection.
Check this for more on the subject:
https://en.wikipedia.org/wiki/Equirectangular_projection

We extended the list of recognized types of cube map (based on the aspect ratio) adding the 2x1 ratio (it needs to be exact ! width == 2 * height) and considering it to be a standard equirect projection.

To test, make sure that by using a skymap that is equirect in the zone, you get the proper result.
This is also possible to use the similar format just for ambient map in the zone.

Since we do a reprojection to cube map when the image is loaded this is slightly slow when using a big texture (>3200).
Quality wise, we don't do any filtering so it's far from perfect but it's fun already!

Here are a few examples of images:
http://paulbourke.net/geometry/transformationprojection/plain_00000.jpg
http://www.panopan.com/sw/jsphere/img/aletsch.jpg 
http://www.ignorancia.org/uploads/panoramas/the-office/sm_office-85.jpg  

And if you own a RICOH THETA the n you can just drop your captures in a zone.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0e13b1623b12bfec6ffe8731d31394ba50d9409a available at: https://gist.github.com/8fe067f2ddebad41f5f293eddc3f1a06<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.dmg

File: libraries/model/src/model/TextureMap.cpp [hunk 1]
```diff
@@ -459,9 +471,107 @@ class CubeLayout {
...
+
+        struct CubeToXYZ {
+            gpu::Texture::CubeFace _face;
+            CubeToXYZ(gpu::Texture::CubeFace face) : _face(face) {}
+
+            glm::vec3 xyzFrom(const glm::vec2& uv) {
+                auto nuv = glm::vec2(-1.0 + 2.0f * uv.x, 1.0 - 2.0f * uv.y);
+
+                switch (_face) {
+                case gpu::Texture::CubeFace::CUBE_FACE_BACK_POS_Z:
```
Comment (ZappoMan) [inline libraries/model/src/model/TextureMap.cpp]: coding standard: this switch statement doesn't match the coding standard... the cases should be indented from the switch.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e8691c2f200c0ebf9c09d2dc80c25f2cc6886ad7 available at: https://gist.github.com/8fe067f2ddebad41f5f293eddc3f1a06<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8f2435daf0d73e2b6f83d1dd6992e3ecf4d3c61b available at: https://gist.github.com/8fe067f2ddebad41f5f293eddc3f1a06<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 91bed86b321379d84f8cfcf784506547df7ed77c available at: https://gist.github.com/8fe067f2ddebad41f5f293eddc3f1a06<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.dmg

**Comment by Nex-Pro** [comment]
##### System: Win 10 desktop 16gb ram Gtx770 2gb

The maps project correctly.
#### @samcake  
Your comment:  *Quality wise, we don't do any filtering so it's far from perfect but it's fun already!*
suggests no filtering but in fact the jpeg artifacts apparent in the images are much reduced when viewed on the zone as compared to viewing in photoshop. 

I tried images up to 13600 x 6800 with no problems. At that size the re-projection only took a few seconds.

Would be nice to be able to set the black point.

### Good work. Pass.

PS. I loaded an image of the Colosseum in Rome and immediately did the noob walk to move closer.  ( *there's a gelateria that sells a pretty decent pistachio nearby.* ) Doh!

**Comment by hifi-gustavo** [comment]
Build console output for commit: b3b2a3511a4eb462deb91da78716a1a9633f94e6 available at: https://gist.github.com/8fe067f2ddebad41f5f293eddc3f1a06<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 91bed86b321379d84f8cfcf784506547df7ed77c available at: https://gist.github.com/8fe067f2ddebad41f5f293eddc3f1a06<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Alpha-PR7761.dmg

**Pull request merged**